apps/api/src/app/events/e2e/bridge-sync.e2e.ts

```diff
@@ -553,4 +559,89 @@ describe('Bridge Sync - /bridge/sync (POST)', async () => {
     const workflowDataWithName = workflowsWithDescription[0];
     expect(workflowDataWithName.description).to.equal('');
   });
+
+  it('should preserve control values across workflow syncs', async () => {
+    const workflowId = 'My Workflow';
+    const spaceSeparatedIdWorkflow = workflow(workflowId, async ({ step }) => {
+      await step.email('send-email', () => ({
+        subject: 'Welcome!',
+        body: 'Hello there',
+      }));
+    });
+    await bridgeServer.start({ workflows: [spaceSeparatedIdWorkflow] });
+
+    const firstSyncResponse = await session.testAgent.post(`/v1/bridge/sync`).send({
+      bridgeUrl: bridgeServer.serverPath,
+    });
+    expect(firstSyncResponse.body.data?.length).to.equal(1);
+
+    const firstWorkflowCountResponse = await workflowsRepository.count({ _environmentId: session.environment._id });
+    expect(firstWorkflowCountResponse).to.equal(1);
+
+    const firstWorkflowResponse = await workflowsRepository.findById(
+      firstSyncResponse.body.data[0]._id,
+      session.environment._id
+    );
+
+    expect(firstWorkflowResponse).to.be.ok;
+    if (!firstWorkflowResponse) {
+      throw new Error('Workflow not found');
+    }
+
+    expect(firstWorkflowResponse.name).to.equal(workflowId);
+    expect(firstWorkflowResponse.type).to.equal(WorkflowTypeEnum.BRIDGE);
+    expect(firstWorkflowResponse.rawData.workflowId).to.equal(workflowId);
+    expect(firstWorkflowResponse.triggers[0].identifier).to.equal(workflowId);
+
+    expect(firstWorkflowResponse.steps.length).to.equal(1);
+    expect(firstWorkflowResponse.steps[0].stepId).to.equal('send-email');
+    expect(firstWorkflowResponse.steps[0]._templateId).to.exist;
+
+    await session.testAgent.put(`/v1/bridge/controls/${workflowId}/send-email`).send({
+      variables: { subject: 'Hello World again' },
+    });
+
+    const firstControlValueResponse = await controlValuesRepository.find({
+      _environmentId: session.environment._id,
+      _workflowId: firstWorkflowResponse._id,
+    });
```

Comment:
Should we perhaps access these controls via API rather than directly to the DB to ensure any business logic between the API endpoint and controls fetching is tested too?

Reply:
due to our current use case e2e tests architecture, IMO the main test here is to `set` control values through API (sync), i would create a separate test for `get` logic where we would definitely need to use the API endpoint.

Reply:
I will do both checks. Good point.